### DISCUSSION TITLE
Strongswan in OpenSuSE Fails to Initiate Session

### DISCUSSION THREAD

**Discussion by blueaxions** [strongswan/strongswan#1904] [Q&A]
Hello,

I'm running strongswan version 5.9.11 on OpenSuSE LEAP 15.4. The installation of strongswan completed without any errors and the strongswan systemd service is running as well. The VPN gateway is Cisco ASA

The issue I'm having is when I try to initiate a session, it fails with the following error message:

2023-09-11T13:09:32.386391-07:00 localhost charon-systemd[31927]: no shared key found for 'VPN_Client_Group' - '192.168.1.10'

Not an expert user of strongswan, so my knowledge of these messages is  limited. I'm assuming its not finding the 'secrets' section somehow?

The /etc/strongswan.d directory has the swanctl.conf file, (attached to this post) and this file has the 'secrets' section for the VPN_Client_Group.

Any help is greatly appreciated.

[messages.txt](https://github.com/strongswan/strongswan/files/12580065/messages.txt)
[swanctl.txt](https://github.com/strongswan/strongswan/files/12580066/swanctl.txt)

**Comment by misch42** [comment]
well, the log is quite clear: "no shared key found for ...". Your need to define keys for the identities you use in your config. strongswan knows the identities "VPN_Client_Group" and "192.168.1.10". Please check if your config states the keys for these identities or perhaps you have a mismatch of the identities. Especiall the "VPN_Client_Group" seems suspicious to me.

Michael

**Comment by blueaxions** [comment]
using the same configuration in strongswan 5.8.2 on a different machine to connect to the same Cisco ASA VPN server works without any issues.

When you say "Please check if your config states the keys for these identities", I'm assuming you're referring to checking the configuration on the Cisco ASA?
If so, I confirmed with the Cisco ASA network admin that the VPN_Client_Group is defined correctly, which makes sense because the same swanctl.conf file works from another machine and is able to establish the tunnel.
Wouldn't that mean the issue is on the initiator side?

**Comment by blueaxions** [comment]
I replaced the real IP address in the log with 192.168.1.10 when sanitizing the logs, I'm unable to share the logs with actual IP addresses in it for security reasons.

**Comment by tobiasbrunner** [comment]
How did you come up with `id=%default` for your PSK? That's clearly neither the local, nor the remote identity. Use one of the actual identities as pointed out by @misch42, or use `id=%any` to associate the key with any identity.

**Comment by blueaxions** [comment]
Thank you @tobiasbrunner and @misch42, changing id=%default to id=VPN_Client_Group fixed the problem.

Thanks for taking the time to help resolve this issue for me, you helped save me a lot of time.

Sincerely, keep up the good work you're doing in the Strongswan community, its invaluable.

-SR
